packages/didcomm/src/modules/connections/services/helpers.ts

```diff
@@ -132,12 +132,14 @@ export function routingToServices(routing: Routing): ResolvedDidCommService[] {
 }
 
 export async function getDidDocumentForCreatedDid(agentContext: AgentContext, did: string) {
+  // Ensure that the DID has been created by us
   const didRecord = await agentContext.dependencyManager.resolve(DidRepository).findCreatedDid(agentContext, did)
-
-  if (!didRecord?.didDocument) {
-    throw new CredoError(`Could not get DidDocument for created did ${did}`)
+  if (!didRecord) {
+    throw new CredoError(`Could not find created did ${did}`)
   }
-  return didRecord.didDocument
+
+  const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+  return await didsApi.resolveDidDocument(did)
```

Comment:
Hmm 🤔 

Are there cases where we don't store our own did?

Reply:
Not really, but I wanted to take all the advantages of `DidsApi.resolveDidDocument()` (such as dynamically generating the did:peer:4 DID Document) making sure that it would only resolve DIDs created by us (I can propagate a filtering parameter but it would modify a good number of methods).

Reply:
This is okay! Just curious :)

Reply:
It also has the benefit of caching, and even DIDs created by us, but not managed by us (e.g. think a did web for which we have one key, but it's stored outside of credo as it's regularly updated)